tools/team-builder/build_teams.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+run(){
+	
+	local OPTIND
+
+  #default values
+	output="fleet_osquery_packages"
+  header=false
+	flags+="--disable-open-folder"
+
+  #Read flags
+	while getopts s:hc:n:p:u:f:d:o:x flag
+	 do
+		case "${flag}" in
+			s) #path to csv file containing team names. Must end with newline char.
+				source=($OPTARG);;
+			h) #indicates whether there is a header included in the file. default false;
+				header=true;;
+			c) #columns in csv file. if not provided, assumes a simple list of names separated by line breaks. not needed if header is provided.
+        set -f 
+        IFS=',' 
+        columns+=($OPTARG) ;; # use the split+glob operator
+			n) #column to use as team name
+				name_column=($OPTARG);;
+			p) #types of installers to create. Pass an individual flag for each type
+				types+=($OPTARG);;
+			u) #Fleet server url
+				url=($OPTARG);;
+			f) #Additional flags to apply to `fleetctl package`
+				flags+=($OPTARG);;
+			d) #include Fleet Desktop
+				flags+="--desktop";;
+			o) #Directory for created packages
+		    output=($OPTARG);;
+      x) #Dry run
+        dry_run="--dry-run";;
+		esac
+	 done
+
+
+	#Verify that passed file exists
+	if !(test -f $source)
```

Comment:
This check doesn't seem to work if no arguments are provided.

```
./build_teams.sh                                                                                                                                                       *[team-builder-script]
here
name
name
./build_teams.sh: line 68: $source: ambiguous redirect
```

I think it's because without quotes around `$source` the shell runs `test -f` (with no argument after `-f`) which always returns 0 (true).

This fixes it:

```suggestion
	if !(test -f "$source")
```

Note that you probably similarly want double-quotes (`"`) around any user-provided variables. This will break again without the quotes if someone provides a filename with a space in it (because bash reads it as multiple separate tokens, rather than one single token with a space in it)

Example with the current script:
```
$ cat "foo bar.txt"
foo
$ ./build_teams.sh "foo bar.txt"
Source file not found
```

Quoting in bash is hard, but generally double quotes are what you want.